### PR TITLE
Return nil for unknown countries and legislatures

### LINKED ISSUE #38: Failing to `find` a Country shouldn't trigger an exception
`Everypolitician::Country.find('Narnia')` should simply return nil, not throw an Exception.

### PR DESCRIPTION
**Pull request by chrismytton**
Rather than throwing an exception for missing countries it seems more ruby-ish to simply return nil.

Fixes #38

### CODE REVIEW

File: lib/everypolitician.rb [hunk 1]
```diff
@@ -74,7 +74,7 @@ def legislature(query)
       legislature = legislatures.find do |l|
         query.all? { |k, v| l.__send__(k).to_s.downcase == v.to_s.downcase }
       end
-      fail Error, "Unknown legislature: #{query}" if legislature.nil?
+      return if legislature.nil?
       legislature
```
Comment (tmtmtmtm) [inline lib/everypolitician.rb]: Isn't this now essentially:

```
foo = do_something
return nil if foo.nil?
return foo
```

i.e.  `do_something`?

**Comment by tmtmtmtm** [comment]
This definitely needs something in the CHANGELOG too (another good use for PR templates…)

File: lib/everypolitician.rb (hunk 1 above)
Comment (chrismytton) [inline lib/everypolitician.rb]: Ha, yep, good point! Fixed in https://github.com/everypolitician/everypolitician-ruby/pull/40/commits/da3906b07c0c82a8e0c72fa1bca31965d8d0a7c5.

**Comment by chrismytton** [comment]
> This definitely needs something in the CHANGELOG too

Ah yep, have added an entry in https://github.com/everypolitician/everypolitician-ruby/pull/40/commits/608318c9a318340f38682bcb910ff81e0e64ad50.

**Comment by chrismytton** [comment]
Fair point about the Rubocop config. The fix in #41 looks like a good way to go to me, so lets get that merged into this branch, then I think this is good to go.

**Comment by chrismytton** [comment]
> Is there something we can (auto-)run to trap things like this being missing?

We could potentially lint the markdown using something like [markdownlint](https://github.com/mivok/markdownlint).

Alternatively we could simply inline the links rather than using reference style, then when people copy and paste the previous version to create a new one they can simply update the link, rather than having to add a new one at the bottom of the file.

**Comment by tmtmtmtm** [comment]
I quite like the idea of doing _both_ of those things!

Keeping the markdown simpler is probably better, but it would also be good to have a linter point out _other_ problems too…

**Pull request merged**